packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe, OwnerManager } from "safe-contracts/Safe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+import { LivenessGuard } from "src/Safe/LivenessGuard.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+// TODO(maurelian): remove me
+import { console2 as console } from "forge-std/console2.sol";
+
+/// @title LivenessModule
+/// @notice This module is intended to be used in conjunction with the LivenessGuard. It should be able to
+///         execute a transaction on the Safe in only a small number of cases.
+contract LivenessModule is ISemver {
+    /// @notice The Safe contract instance
+    Safe public immutable safe;
+
+    /// @notice The LivenessGuard contract instance
+    ///         This can be updated by replacing with a new module and switching out the guard.
+    LivenessGuard public immutable livenessGuard;
+
+    /// @notice The interval, in seconds, during which an owner must have demonstrated liveness
+    ///         This can be updated by replacing with a new module.
+    uint256 public immutable livenessInterval;
+
+    /// @notice The minimum number of owners before ownership of the safe is transferred to the fallback owner.
+    ///         This can be updated by replacing with a new module.
+    uint256 public immutable minOwners;
+
+    /// @notice The fallback owner of the Safe
+    ///         This can be updated by replacing with a new module.
+    address public immutable fallbackOwner;
+
+    /// @notice The storage slot used in the safe to store the guard address
+    ///         keccak256("guard_manager.guard.address")
+    uint256 internal constant GUARD_STORAGE_SLOT = 0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8;
+
+    /// @notice The address of the first owner in the linked list of owners
+    address internal constant SENTINEL_OWNERS = address(0x1);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    // Constructor to initialize the Safe and baseModule instances
+    constructor(
+        Safe _safe,
+        LivenessGuard _livenessGuard,
+        uint256 _livenessInterval,
+        uint256 _minOwners,
+        address _fallbackOwner
+    ) {
+        safe = _safe;
+        livenessGuard = _livenessGuard;
+        livenessInterval = _livenessInterval;
+        minOwners = _minOwners;
+        fallbackOwner = _fallbackOwner;
+    }
+
+    /// @notice This function can be called by anyone to remove an owner that has not signed a transaction
+    ///         during the liveness interval. If the number of owners drops below the minimum, then the
+    ///         ownership of the Safe is transferred to the fallback owner.
+    function removeOwner(address owner) external {
+        // Check that the guard has not been changed
+        _verifyGuard();
```

Comment:
We verify guard inside of `_verifyFinalState` at the end of this call, is it necessary to do it at the beginning as well?

Reply:
no, I'll remove this.